VmAgent.Core/Model/MonitoringPersistedState.cs

```diff
@@ -12,13 +12,15 @@ public enum MonitoringApplicationState
         Installed,
         Executing,
         ExecutionFailure,
+        SignaledToStop,
         ExecutionComplete
     }
     public class MonitoringPersistedState
     {
         public int MonitoringPID { get; set; }
-        public int RetryCount { get; set; }
         public MonitoringApplicationState MonitoringState { get; set; }
-        public DateTime ApplicationStartTime { get; set; } 
+        public DateTime ExpectedCompletionTime { get; set; } 
+        public string CallerSessionID { get; set; }
```

Comment:
Minor: I think we have been using "Id" instead of "ID" in most places. Consider updating this and MonitoringPID above